### ISSUE TITLE
Upgrade to 3.1.0 breaks block editor

### ISSUE THREAD

**Issue by claytoncollie** [markjaquith/page-links-to#73]
Just hit update on 3.1.0, then went to add a new object inside a custom post type. Screen was all white except for the stadnard Gutenberg buttons to recover, copy text and copy error. Tried to deactivate then re-activate the plugin with no luck.

```
    TypeError: Cannot read property '_links_to' of undefined
    at LinksTo.getUrl (https://frosszelnick.wpengine.com/wp-content/plugins/page-links-to/js/gutenberg.min.js?ver=3.1.0:53:27)
    at LinksTo.hasUrl (https://frosszelnick.wpengine.com/wp-content/plugins/page-links-to/js/gutenberg.min.js?ver=3.1.0:65:16)
    at new LinksTo (https://frosszelnick.wpengine.com/wp-content/plugins/page-links-to/js/gutenberg.min.js?ver=3.1.0:46:31)
    at Yf (https://frosszelnick.wpengine.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.8.4:67:42)
    at qg (https://frosszelnick.wpengine.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.8.4:92:302)
    at hi (https://frosszelnick.wpengine.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.8.4:104:285)
    at Qg (https://frosszelnick.wpengine.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.8.4:144:217)
    at Rg (https://frosszelnick.wpengine.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.8.4:145:76)
    at Sc (https://frosszelnick.wpengine.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.8.4:158:109)
    at Kc (https://frosszelnick.wpengine.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.8.4:155:16)
```

I'm running WP 5.2 with Genesis 2.10.1 on PHP 7.2 and MySQL 5.6.34

**Comment by KZeni** [comment]
I've also encountered this issue. I'm also only experiencing it with the Gutenberg editor on a custom post type, currently. As an aside, this custom post type is was implemented using Custom Post Type UI (https://wordpress.org/plugins/custom-post-type-ui/), if that matters.

**Comment by markjaquith** [comment]
Working on this right now! I was able to reproduce the issue with a new custom post type I created. For some reason, this CPT is not getting meta passed to it, thus the error about `this.props.meta`.

**Comment by markjaquith** [comment]
Annoyingly, if your post type doesn't declare support for `custom-fields`, post meta fields aren't available to Gutenberg.

I think I'm going to solve it by saying that if your post type supports PLT and Gutenberg (`editor`), then you get `custom-fields` support for that post type.

```php
                foreach ( $post_type_names as $type ) {
                        if ( self::is_supported_post_type( $type ) ) {

                               // ========== NEW CODE ========== 
                               if ( post_type_supports( $type, 'editor' ) ) {
                                       add_post_type_support( $type, 'custom-fields' ); // The Block Editor REQUIRES this.
                               }
                               // ========= / NEW CODE ========== 

                                $this->register_meta( self::LINK_META_KEY, $type );
                                $this->register_meta( self::TARGET_META_KEY, $type );
                        }
                }
```

**Comment by markjaquith** [comment]
A more surgical method is to hook in really late on `use_block_editor_for_post`:

```php
$this->hook( 'use_block_editor_for_post', 99999 );
```

With this:

```php
public function use_block_editor_for_post( $use_block_editor, $post ) {
	if ( $use_block_editor && self::is_supported_post_type( get_post_type( $post ) ) ) {
		add_post_type_support( get_post_type( $post ), 'custom-fields' );
	}

	return $use_block_editor;
}
```

**Comment by markjaquith** [comment]
Thanks @claytoncollie and @KZeni! This fixes it, for me. New release will be going out soon, after I take a look at a couple other issues.

**Comment by claytoncollie** [comment]
thanks @markjaquith !!!

**Comment by KZeni** [comment]
I've checked it out in the 3.1.1 version update, and things are looking better for me now. Thank you!

As an aside, it took a moment to realize it's now labelled as "Custom Permalink" as an inline attribute instead of "Page Links To" being a more formal section. Might be good to get some screenshots updated on the WP.org plugin listing to show where this is now found in the new editor. Also, I did have a thought that maybe this could be "_singular post type label_ Links To" ("Page Links To", "Product Links To", etc.) to help continue the "[...] Links To" convention to keep it familiar & consistent while it's still made to be a bit more contextual. Just a thought.

Thanks again for getting this fixed!
